src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -38,11 +46,19 @@ public class WebJobsScriptHostService : IHostedService, IScriptHostManager, IDis
             _rootServiceProvider = rootServiceProvider ?? throw new ArgumentNullException(nameof(rootServiceProvider));
             _rootScopeFactory = rootScopeFactory ?? throw new ArgumentNullException(nameof(rootScopeFactory));
             _applicationHostOptions = applicationHostOptions ?? throw new ArgumentNullException(nameof(applicationHostOptions));
-            _scriptWebHostEnvironment = scriptWebHostEnvironment ?? throw new ArgumentNullException(nameof(scriptWebHostEnvironment));
+            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _performanceManager = hostPerformanceManager ?? throw new ArgumentNullException(nameof(hostPerformanceManager));
+            _healthMonitorOptions = healthMonitorOptions ?? throw new ArgumentNullException(nameof(healthMonitorOptions));
 
             _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
 
             State = ScriptHostState.Default;
+
+            if (ShouldMonitorHostHealth)
```

Comment:
This code is basically being ported from v1 ScriptHostManager - behaviorally it is the same